src/wallet/wallet.cpp

```diff
@@ -2048,11 +2048,15 @@ TransactionError CWallet::SignPSBT(PartiallySignedTransaction& psbtx, bool& comp
             txin.assetIssuance.nAmount = input.m_issuance_value_commitment;
         } else if (input.m_issuance_value) {
             txin.assetIssuance.nAmount.SetToAmount(*input.m_issuance_value);
+        } else {
```

Comment:
These variables were never cleared, so they retained the value of the previous loop iteration, which resulted in problems when there were multiple inputs beyond the first issuance, as well as problems when reissuing assets (where there are no inflation keys created).